namer/namer.cc

```diff
@@ -1049,6 +1094,44 @@ class SymbolDefiner {
         }
     }
 
+    core::FieldRef insertField(core::MutableContext ctx, const core::FoundField &field) {
```

Comment:
Does this method actually ever get called? It looks like this is only called from `defineNonDeletableSingle`, but the only place where we call `defineNonDeletableSingle` has a check that says `ref.kind() != core::FoundDefinitionRef::Kind::Field`.

Is this PR a no-op?

Should we also make sure that we delete the logic from resolver that calls `enterFieldSymbol`/`enterStaticFieldSymbol` so that we can be sure it's being powered by this change alone?

Reply:
We now set `resultType` directly on a pre-existing field ("set type directly on todo fields").  I guess we should be able to go further and just `ENFORCE` that the field is there after `findMember` (or making entering the "uncommon" case inside a conditional) instead?

(I suppose it's also possible that we have a pre-existing thing that is *not* a field, and we'd still need the `enter` variants for that case.)